text_to_image/stable_diffusion/run.py

```diff
@@ -0,0 +1,95 @@
+    from utils.text_to_image.stable_diffusion import StableDiffusion
+    from text_to_image.stable_diffusion.stablediffusion.ldm.models.diffusion.ddim import DDIMSampler
+    from text_to_image.stable_diffusion.stablediffusion.scripts.txt2img import load_model_from_config
+
+    seed_everything(42)
+    config = OmegaConf.load(f"{config}")
+    model = load_model_from_config(config, f"{model_path}", torch.device("cpu"))
+    sampler = DDIMSampler(model, device=torch.device("cpu"))
+    shape = [4, 512 // 8, 512 // 8]
+
+    # TODO: torchscript stuff, should it stay here?
+    unet = model.model.diffusion_model
+    decoder = model.first_stage_model.decoder
+
+    stablediffusion_data = pathlib.Path(get_downloads_path(), "stable_diffusion")
+    unet_path = Path(stablediffusion_data, "unet.pt")
+    decoder_path = Path(stablediffusion_data, "decoder.pt")
+    if not stablediffusion_data.exists():
+        stablediffusion_data.mkdir(exist_ok=True)
+
+    with torch.no_grad(), nullcontext():
+        if unet_path.exists():
+            scripted_unet = torch.jit.load(unet_path)
+        else:
+            img_in = torch.ones(2, 4, 96, 96, dtype=torch.float32)
+            t_in = torch.ones(2, dtype=torch.int64)
+            context = torch.ones(2, 77, 1024, dtype=torch.float32)
+            scripted_unet = torch.jit.trace(unet, (img_in, t_in, context))
+            scripted_unet = torch.jit.freeze(scripted_unet)
+            torch.jit.save(scripted_unet, unet_path)
+        model.model.scripted_diffusion_model = scripted_unet
+
+        if decoder_path.exists():
+            scripted_decoder = torch.jit.load(decoder_path)
+        else:
+            samples_ddim = torch.ones(1, 4, 96, 96, dtype=torch.float32)
+            scripted_decoder = torch.jit.trace(decoder, (samples_ddim))
+            scripted_decoder = torch.jit.freeze(scripted_decoder)
+            torch.jit.save(scripted_decoder, decoder_path)
+        model.first_stage_model.decoder = scripted_decoder
+
+    def single_pass_pytorch(_runner, _stablediffusion):
+        _runner.run(batch_size * steps)
+        _stablediffusion.submit_count()
+
+    def wrapper():
+        with torch.no_grad(), nullcontext(torch.device("cpu")), model.ema_scope():
+            uc = model.get_learned_conditioning(batch_size * [""]) if scale != 1.0 else None
+            samples, _ = sampler.sample(S=steps,
+                                        conditioning=model.get_learned_conditioning([prompt] * batch_size),
+                                        batch_size=batch_size,
+                                        shape=shape,
+                                        verbose=False,
+                                        unconditional_guidance_scale=scale,
+                                        unconditional_conditioning=uc,
+                                        eta=0.0,
+                                        x_T=None)
+
+    runner = PyTorchRunnerV2(wrapper)
```

Comment:
You could pass sampler.sample func here (without executing it) instead of wrapper. Then pass all variables (S=steps, conditioning, ...) in single_pass_pytorch func after task_size, i.e.: _runner.run(batch_size * steps, S=steps, conditioning=...., ....)

Reply:
What about:

```
x_samples = model.decode_first_stage(samples)
x_samples = torch.clamp((x_samples + 1.0) / 2.0, min=0.0, max=1.0)

 for x_sample in x_samples:
    x_sample = 255. * rearrange(x_sample.cpu().numpy(), 'c h w -> h w c')
    img = Image.fromarray(x_sample.astype(np.uint8))
```

Full workload is from text to image.

Reply:
in you second message do you suggest to save images? I don't do that at the moment because I don't measure the accuracy and thus I think it would be irrelevant for this PR.